src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -78,6 +109,50 @@ class DataCube {
             );
     }
 
+    filterTemporal(extent, dim) {
+        const dimensionName = dim ? dim : this.temporal_dimension_name;
```

Comment:
> If no specific dimension is specified or it is set to null, the filter applies to all temporal dimensions

So there could be multiple dimensions of type `TEMPORAL`, or there could be one but not the default one (that goes by the name `temporal_dimension_name`).

Reply:
> So there could be multiple dimensions of type TEMPORAL, or there could be one but not the default one (that goes by the name temporal_dimension_name).

We handle the second case, but not the first one. I have somewhat a hard time imagining how this would work in practice. But technically I see no problem of iterating over all temporal dimensions and applying the same algorithm to all dimensions.

Reply:
> We handle the second case

I don't think we do, unless I'm missing something. 

> I have somewhat a hard time imagining how this would work in practice

Yes, it's definitely not a very common scenario, but we should still follow specs and this should be achieved pretty simply, as you mentioned. I think the most likely scenario is that the default temporal dimension gets renamed and `filter_temporal` is called with `dim=null`.

Reply:
Ok, we weren't entirely clear.

> I think the most likely scenario is that the default temporal dimension gets renamed and filter_temporal is called with dim=null.

We definitely need to address this case, I missed this one, will update. (Currently `dim` would have to be set accordingly for this to work).

> I have somewhat a hard time imagining how this would work in practice. But technically I see no problem of iterating over all temporal dimensions and applying the same algorithm to all dimensions.

This was referring to having multiple dimensions. Where we would then iterate over all of them. Is this relevant?

Reply:
Funny enough, there was even an issue to clarify this by Anže:

https://github.com/Open-EO/openeo-processes/issues/203

Reply:
Cool, this issue confirms clearly that it should indeed be applied to all temporal dimensions 👍

Reply:
Looks good.